src/components/useMeeting.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  Meeting,
+  MeetingSeries,
+  TalkingPoint,
+  TalkingPointTicket,
+} from "../types";
+import { useEffect, useState } from "react";
+import { getSeries, saveTalkingPoint } from "../server";
+
+export function useMeeting() {
+  const [series, setSeries] = useState<MeetingSeries>();
+  const [currentMeeting] = series?.meetings || [];
+
+  useEffect(() => {
+    // consider getSeries an apollo query
+    getSeries().then(setSeries);
+    // We don't refetch, after the initial load we only rely on state updates
+  }, []);
+
+  const updateMeeting = (updatedMeeting: Meeting) => {
+    const meetings = series!.meetings!.map((meeting) =>
+      meeting.id === meeting.id ? updatedMeeting : meeting
+    );
+
+    setSeries({ ...series!, meetings });
+  };
+
+  const updateTalkingPoint = (talkingPoint: TalkingPoint) => {
+    const agendaItems = currentMeeting.agendaItems.map((item) =>
+      talkingPoint.id === item.id ? talkingPoint : item
+    );
+
+    updateMeeting({
+      ...currentMeeting,
+      agendaItems,
+    });
+
+    /*
+      If this fails we have some options:
+      - block the item until the transaction is finished (super bad in my opinion)
+      - undo data and ask user to perform action again (super bad in my opinion)
+      - retry the same request until it works and cancel if same item receives an update (apollo has some ways to handle this)
+      - send the full meeting to the backend. it will fix all inconsistencies, since the ui now has the stable data about the meeting.
+    */
+    saveTalkingPoint(talkingPoint);
+  };
```

Comment:
Yes, as mentioned in the previous documents. Optimistic UI is the way to go, otherwise, we would need to block the user for every action to keep consistency in the front end. Once we have consistency in the frontend state it is easier to fix something if a request fails for example